UM-USElab/gradecraft-development
# 1245 Student Created Badges

### Status
**IN DEVELOPMENT**

### Description
As a student
I should be able to 'propose' badges,
Including setting their names, graphics, and descriptions.

As an instructor
I should be notified that there are student-created badges to review
And be able to edit their elements before approving or rejecting

The badge should retain info regarding who created it, and if they were created by a student
This should be highlighted in the interface to both students and instructors.

### Related PRs
N/A

### Todos

### Deploy Notes
N/A

### Gem dependencies
N/A

### Migrations
YES

### Steps to Test or Reproduce
As a student, navigate to the modified 'Badges' page. I will see a new button "Propose Badge". Clicking on the button will take me to the New Badge Form where I can submit a new badge to the instructor of the course.

As an instructor, when I navigate to the 'Badges' page, I will now see 3 tabs: 'Accepted', 'Proposed', and 'Rejected'. By navigating to the show page of one of the proposed badges I will be able to see who submitted the proposed badge. I will also be able to Accept of Reject the badge by using the Options Gear on the show page. Accepting or Rejecting will move it to the appropriate tab on the instructor badges page.

By default, when an instructor creates a badge, it will be set to 'Accepted'

### Impacted Areas in Application
List general components of the application that this PR will affect:

* Badges

======================
Closes #1245
